lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -124,23 +130,39 @@ export default function getRecipeInterface(
             sessionData?: any;
             userContext: any;
         }): Promise<Session> {
-            const disableAntiCSRF = config.getTokenTransferMethod({ req, userContext }) === "header";
+            logDebugMessage("createNewSession: Started");
+            let outputTransferMethod = config.getTokenTransferMethod({ req, userContext });
+            logDebugMessage("createNewSession: got transfer method " + outputTransferMethod);
+            if (outputTransferMethod === "missing_auth_header") {
+                outputTransferMethod = "header";
+            }
+            logDebugMessage("createNewSession: using transfer method " + outputTransferMethod);
+
+            const disableAntiCSRF = outputTransferMethod === "header";
             let response = await SessionFunctions.createNewSession(
                 helpers,
                 userId,
                 disableAntiCSRF,
                 accessTokenPayload,
                 sessionData
             );
-            attachCreateOrRefreshSessionResponseToExpressRes(config, req, res, response, userContext);
+
+            for (const transferMethod of availableTokenTransferMethods) {
+                if (transferMethod !== outputTransferMethod) {
+                    clearSession(config, req, res, userContext, transferMethod);
```

Comment:
this will add remove to the front-token - is that OK? I know that below we are calling `attachCreateOrRefreshSessionResponseToExpressRes` which should overwrite that `remove` value, but isn't this a bit strange of a code flow?

Im ok with this though - just thought to comment